lib/presentation/features/home/home_screen.dart

```diff
@@ -26,11 +25,18 @@ class HomeScreen extends StatelessWidget {
             Widget child = const Center(
               child: CircularProgressIndicator(),
             );
-            if (state.loading) return child;
-            if (state.error == null && !state.loading) {
-              child = ListCharacters(list: state.characters);
+            if (state.characters != null && state.error == null) {
+              child = ListCharacters(list: state.characters, error: state.error, hasReachedMax: state.hasReachedMax);
+            } else if (state.characters == null && state.error == null) {
+              return child;
+            } else if (state.characters == null && state.error != null) {
```

Comment:
I would recomend simplify it by early return:
if (state.characters != null && state.error == null) return ...
if (state.characters == null && state.error == null) return ...
...

It is easier to read.

Reply:
done